scanomatic/ui_server_data/js/som/experiment.js

```diff
@@ -0,0 +1,376 @@
+  } else {
+    sib.attr('src', '/images/yeastNOK.png').attr('alt', tooltip).attr('title', tooltip);
+  }
+}
+
+function getDurationAsMinutes() {
+  return (((duration[0] * 24) + duration[1]) * 60) + duration[2];
+}
+
+export function validateExperiment() {
+  setValidationStatus(
+    '#project-path',
+    projectPathValid,
+    projectPathValid ? 'Everything is cool' : projectPathInvalidReason,
+  );
+  const scanner = $('#current-scanner');
+  const scannerOK = scanner.val() !== '' && scanner.val() !== undefined && scanner.val() != null;
+  setValidationStatus(scanner, scannerOK, scannerOK ? 'The scanner is yours' :
+    'It would seem likely that you need a scanner... to scan');
+  const activePlates = fixturePlates.filter(e => e.active).length > 0;
+  let tooltip;
+  if (!fixtureSelected) {
+    tooltip = 'Select fixture';
+  } else {
+    tooltip = activePlates ? 'No plates active would mean doing very little' : 'All good!';
+  }
+  setValidationStatus('#current-fixture', fixtureSelected && activePlates, tooltip);
+  const timeMakesSense = getDurationAsMinutes() >= 0;
+  setValidationStatus(
+    '#project-duration',
+    timeMakesSense,
+    timeMakesSense ? "That's a splendid project duration" : "Nope, you can't reverse time",
+  );
+  InputEnabled(
+    $('#submit-button'),
+    projectPathValid && scannerOK && fixtureSelected && activePlates && timeMakesSense,
+  );
+}
+
+export function setPoetry(input) {
+  $(input).html(poetry).css('font-style', 'italic');
+
+  $(input).on('focus', function handleFocus() {
+    if ($(this).html() === poetry) {
+      $(this).html('').css('font-style', 'normal');
+    }
+  });
+
+  $(input).on('blur', function handleBlur() {
+    if ($(this).html() === '') {
+      $(this).html(poetry).css('font-style', 'italic');
+    }
+  });
+}
+
+export function setExperimentRoot(input) {
+  getPathSuggestions(
+    input,
+    true,
+    '',
+    null,
+    (data) => {
+      projectPath = $(input).val();
+      projectPathValid = data.valid_parent && !data.exists;
+      projectPathInvalidReason = data.reason;
+
+      setValidationStatus(
+        '#project-path',
+        projectPathValid,
+        projectPathValid ? 'Everything is cool' : projectPathInvalidReason,
+      );
+      if (data.valid_parent && !data.exists) {
+        $('#experiment-title').html(`Start Experiment '${data.prefix}'`);
+      } else {
+        $('#experiment-title').html('Start Experiment');
+      }
+    },
+  );
+}
+
+function getDescription(index, description) {
+  return `${"<div class='plate-description'>" +
+        "<input type='hidden' value='"}${index}'>` +
+        `<label for='plate-description-${index}'>Plate ${index}</label>` +
+        `<input class='long' id='plate-description-${index}' value='${description
+        }' placeholder='Enter medium and, if relevant, what experiment this plate is.' onchange='cacheDescription(this);'></div>`;
```

Comment:
this onchange should probably be `som.cache...`

Reply:
Fixed.